src/lightning_utilities/core/enums.py

```diff
@@ -67,12 +49,17 @@ def from_str(
             for enum_key, enum_val in cls.__members__.items():
                 if enum_val == value:
                     return cls[enum_key]
+        raise ValueError(f"Invalid match: expected one of {cls._allowed_matches(source)}, but got {value}.")
 
-        warnings.warn(UserWarning(f"Invalid string: expected one of {allowed}, but got {value}."))
-        return None
+    @classmethod
+    def maybe_from_str(cls, value: str, source: Literal["key", "value", "any"] = "key") -> Optional["StrEnum"]:
```

Comment:
this looks strange :confused: I would really go with strict flag

Reply:
See https://adamj.eu/tech/2021/07/10/python-type-hints-how-to-avoid-the-boolean-trap/ why booleans like this are an antipattern

Feel free to suggest a different name